File: frontend/src/types/job.ts
export interface Company {
  id: string;
  name: string;
  normalized_name: string;
  website?: string | null;
  logo_url?: string | null;
  description?: string | null;
  engineering_focus?: string | null;
  created_at: string;
  updated_at: string;
}

export interface Job {
  id: string;
  company_id?: string | null;
  slug: string;
  title: string;
  sanitized_title?: string | null;
  company_name: string;
  location: string;
  remote: boolean;
  url: string;
  description: string;
  tags: string[];
  job_types: string[];
  salary_min?: number | null;
  salary_max?: number | null;
  salary_currency?: string | null;
  published_at: string;
  content_hash: string;
  company?: Company | null;
  created_at: string;
  updated_at: string;
}

export interface JobFilterParams {
  keywords?: string;
  location?: string;
  remote_only?: boolean;
  job_type?: string;
  min_salary?: number;
  limit?: number;
  offset?: number;
}

File: backend/app/schemas/job.py
"""Job and Discovery Feed Pydantic schemas."""

import uuid
from datetime import datetime
from decimal import Decimal
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, ConfigDict, Field


class JobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    slug: str
    title: str
    sanitized_title: Optional[str] = None
    company_name: str
    location: str
    remote: bool
    url: str
    description: str
    tags: List[str] = Field(default_factory=list)
    job_types: List[str] = Field(default_factory=list)
    salary_min: Optional[Decimal] = None
    salary_max: Optional[Decimal] = None
    salary_currency: Optional[str] = "EUR"
    published_at: datetime
    created_at: Optional[datetime] = None
    source_name: Optional[str] = "Arbeitnow"


class PaginationMeta(BaseModel):
    total_items: int
    total_pages: int
    current_page: int
    limit: int


class PaginatedJobsResponse(BaseModel):
    items: List[JobRead]
    pagination: PaginationMeta


class JobSyncRequest(BaseModel):
    source: str = "arbeitnow"
    force_refresh: bool = False


class JobSyncResponse(BaseModel):
    status: str
    source: str
    fetched_count: int
    new_jobs_inserted: int
    duplicates_skipped: int
